crowbar_framework/app/models/crowbar_service.rb

```diff
@@ -320,7 +320,7 @@ def prepare_nodes_for_crowbar_upgrade
     not_ready_for_upgrade = []
     all_nodes = NodeObject.all
     all_nodes.each do |n|
-      next if n.admin? && ["ready", "crowbar_upgrade"].include?(n.state)
+      next unless n.admin? && ["ready", "crowbar_upgrade"].include?(n.state)
```

Comment:
right... confusing because we changed the logic completely with the introduction of `next`